src/tscore/ink_queue.cc

```diff
@@ -94,6 +94,12 @@ static const ink_freelist_ops *default_ops = &freelist_ops;
 static ink_freelist_list *freelists                = nullptr;
 static const ink_freelist_ops *freelist_global_ops = default_ops;
 
+inline void
+dummy_forced_read(void *mem)
+{
+  static_cast<void>(*const_cast<int volatile *>(reinterpret_cast<int *>(mem)));
```

Comment:
I prefer code that's less rather than more obscure.  In any case, seems misleading to say something is maybe unused when it's definitely unused.

We should never miss a chance to use a static_cast apparently.

It's dummy in the sense that we are reading a memory location as some sort of fragile, ratchet, half-assed memory fence, rather than to read the value of a memory location.  Instead of using the more portable code I wrote to replace this mess.

Reply:
`[[maybe_unused]]` is the official replacement in C++17 for casting to `void` to avoid discarding computation due to optimization.

Reply:
cppreference.com as much as I does not seem to have gotten the memo you got about `[[maybe_unused]]`:  https://en.cppreference.com/w/cpp/language/attributes/maybe_unused